code/game/machinery/camera/camera_assembly.dm

```diff
@@ -105,6 +105,8 @@
 	if(is_type_in_list(W, possible_upgrades) && !is_type_in_list(W, upgrades)) // Is a possible upgrade and isn't in the camera already.
 		if(!user.unEquip(W))
 			return
+		if(!W.drop_item()) // no eating borg items
+			return

```

Comment:
still wrong. But it doesn't matter anyway because of the lines just above: the unEquip() check doesn't catch the borg analyzer, which means the borg analyzer is likely missing the NODROP flag which is the real issue here.